test/error_tracker/telemetry_test.exs

```diff
@@ -11,22 +11,36 @@ defmodule ErrorTracker.TelemetryTest do
   end
 
   test "events are emitted for new errors" do
+    {exception, stacktrace} =
+      try do
+        raise "This is a test"
+      rescue
+        e -> {e, __STACKTRACE__}
+      end
+
     # Since the error is new, both the new error and new occurrence events will be emitted
-    report_error(fn -> raise "This is a test" end)
+    %Occurrence{error: error = %Error{}} = ErrorTracker.report(exception, stacktrace)
     assert_receive {:telemetry_event, [:error_tracker, :error, :new], _, %{error: %Error{}}}
 
     assert_receive {:telemetry_event, [:error_tracker, :occurrence, :new], _,
                     %{occurrence: %Occurrence{}}}
 
     # The error is already known so the new error event won't be emitted
-    report_error(fn -> raise "This is a test" end)
+    ErrorTracker.report(exception, stacktrace)
 
-    refute_receive {:telemetry_event, [:error_tracker, :error, :new], _,
-                    %{occurrence: %Occurrence{}}},
+    refute_receive {:telemetry_event, [:error_tracker, :error, :new], _, %{error: %Error{}}},
```

Comment:
This was wrong. We were checking the telemetry event incorrectly and it did never match because it has an `%Error{}` instead of an `%Occurrence{}`.

The reason is that every time we raised the error in a different line, so they were different errors and not the same.

The new code ensures that the same error is raised multiple times and helped me catch this error.

Reply:
Awesome!